src/bootstrap/src/utils/shared_helpers.rs

```diff
@@ -60,16 +64,30 @@ pub fn exe(name: &str, target: &str) -> String {
 /// If it was not defined, returns 0 by default.
 ///
 /// Panics if "RUSTC_VERBOSE" is defined with the value that is not an unsigned integer.
+#[inline(always)]
 pub fn parse_rustc_verbose() -> usize {
-    match env::var("RUSTC_VERBOSE") {
-        Ok(s) => usize::from_str(&s).expect("RUSTC_VERBOSE should be an integer"),
-        Err(_) => 0,
+    static mut VERBOSITY: usize = 0;
+    static INIT: Once = Once::new();
+
+    // SAFETY: Write only happens once.
+    unsafe {
+        // Result of this function never changes, so cache it to avoid recomputing
+        // it repeatedly as it’s called thousands of times from the rustc shim.
+        INIT.call_once(|| {
+            VERBOSITY = match env::var("RUSTC_VERBOSE") {
+                Ok(s) => usize::from_str(&s).expect("RUSTC_VERBOSE should be an integer"),
+                Err(_) => 0,
+            };
+        });
+
+        VERBOSITY
     }
```

Comment:
```suggestion
    static VERBOSITY: LazyLock<usize> = LazyLock::new(|| {
        env::var("RUSTC_VERBOSE").map_or(0, |s| {
            s.parse().expect("RUSTC_VERBOSE should be an integer")
        })
    });

    *VERBOSITY
```

Reply:
I intentionally preferred `Once` because I thought it was the cheapest way, am I wrong on this?

Reply:
I guess it's slightly cheaper than `LazyLock` as it generates fewer instructions. I've checked Godbolt and run a micro benchmark for this function but couldn't come to a good conclusion wrt. performance difference (likely case: my benchmark sucked).